test/routes/src/server/routes/t_launchpad.js

```diff
@@ -74,113 +74,163 @@ describe('The Launchpad API endpoint', () => {
       });
     });
 
-    context('when snap does not exist', () => {
-      const caveatId = 'dummy caveat';
+    context('when repo contains valid snapcraft.yaml', () => {
```

Comment:
Quite a lot of changes here due to moving tests around.

I moved some tests that shared same mocked requests (valid snapcraft.yaml, name registered) into common nested contexts to avoid duplicating same calls to `nock`.